docs/vec_of_option.md

```diff
@@ -0,0 +1,212 @@
+            let mut steps = 1 as IdxSize;
+            for next in iter.by_ref() {
+                steps += 1;
+                if let Some(high) = next {
+                    let steps_n: T::Native = NumCast::from(steps).unwrap();
+                    interpolation_branch(low, high, steps, steps_n, &mut out);
+                    out.push(high);
+                    low = high;
+                    break;
+                }
+            }
+        }
+    }
+    if first != 0 || last != chunked_arr.len() {
+        let mut validity = MutableBitmap::with_capacity(chunked_arr.len());
+        validity.extend_constant(chunked_arr.len(), true);
+
+        for i in 0..first {
+            validity.set(i, false);
+        }
+
+        for i in last..chunked_arr.len() {
+            validity.set(i, false);
+            out.push(Zero::zero())
+        }
+
+        let array = PrimitiveArray::new(
+            T::get_dtype().to_arrow(CompatLevel::newest()),
+            out.into(),
+            Some(validity.into()),
+        );
+        ChunkedArray::with_chunk(chunked_arr.name(), array)
+    } else {
+        ChunkedArray::from_vec(chunked_arr.name(), out)
+    }
+}
+```
+
+That's a lot to digest at once, so let's take small steps and focus on the core logic.
+At the start, we store the indexes of the first and last null values:
+
+```rust
+let first = chunked_arr.first_non_null().unwrap();
+let last = chunked_arr.last_non_null().unwrap() + 1;
+```
+
+Then, we push actual zeroes to the output `Vec` - as many as there are nulls before the first valid value in the input:
+
+```rust
+let mut out = Vec::with_capacity(chunked_arr.len());
+for _ in 0..first {
+    out.push(Zero::zero());
+}
+```
+
+We skip the first `first` elements and start interpolating, but the droids we're looking for are not here, so let's skip that part:
+
+```rust
+let mut iter = chunked_arr.iter().skip(first);
+let mut low = iter.next().unwrap().unwrap();
+out.push(low);
+while let Some(next) = iter.next() {
+    // Interpolation logic
+}
+```
+
+Now, after _most_ of the work is done, we find something interesting - see the comments:
+
+```rust
+if first != 0 || last != chunked_arr.len() {
+    // A validity mask is created for the vector, initially all set to true
+    let mut validity = MutableBitmap::with_capacity(chunked_arr.len());
+    validity.extend_constant(chunked_arr.len(), true);
+
+    for i in 0..first {
+        // The indexes corresponding to the zeroes before the first valid value
+        // are set to false (invalid)
+        validity.set(i, false);
+    }
+
+    for i in last..chunked_arr.len() {
+        // The indexes corresponding to the values after the last valid value
+        // are set to false (invalid)
+        validity.set(i, false);
+
+        out.push(Zero::zero())  // This is equivalent to the zeroes pushed
+                                // before the first valid value, it's just done
+                                // out of order
```

Comment:
not sure I understand the comment, how about "put zeros to the end of the vector after the last valid value"?

Reply:
It's written like that because there's a comment on the zeroes pushed before the first valid value as well, but it's done before, in a block above.

Reply:
I'll rewrite it though, it's a little strange indeed